Gems/PhysX/Code/Source/JointComponent.cpp

```diff
@@ -140,22 +139,27 @@ namespace PhysX
 
         if (m_configuration.m_leadEntity.IsValid())
         {
-            Physics::RigidBodyRequestBus::EventResult(info.m_leadBody
-                , m_configuration.m_leadEntity
-                , &Physics::RigidBodyRequests::GetRigidBody);
-
-            if (!info.m_leadBody)
+            AzPhysics::SimulatedBodyComponentRequestsBus::EventResult(
+                info.m_leadBody, m_configuration.m_leadEntity, &AzPhysics::SimulatedBodyComponentRequests::GetSimulatedBody);
+
+            // Report a warning if there's no lead body or the body type is not one of the supported.
+            // In the future only body type validation will be needed 
+            if (!info.m_leadBody ||
+                !(info.m_leadBody->GetNativeType() == NativeTypeIdentifiers::RigidBody ||
+                 info.m_leadBody->GetNativeType() == NativeTypeIdentifiers::RigidBodyStatic))
             {
-                const AZStd::string entityWithoutBodyWarningMsg("Rigid body not found in lead entity associated with joint. Joint treated as constraint on global position.");
+                info.m_leadBody = nullptr;
+                const AZStd::string entityWithoutBodyWarningMsg(
+                    "Simulated body not found in lead entity associated with joint. Joint treated as constraint on global position.");
                 WarnInvalidJointSetup(m_configuration.m_leadEntity, entityWithoutBodyWarningMsg);
             }
         }
 
-        Physics::RigidBodyRequestBus::EventResult(info.m_followerBody
-            , m_configuration.m_followerEntity
-            , &Physics::RigidBodyRequests::GetRigidBody);
+        AzPhysics::SimulatedBodyComponentRequestsBus::EventResult(
+            info.m_followerBody, m_configuration.m_followerEntity, &AzPhysics::SimulatedBodyComponentRequests::GetSimulatedBody);
 
-        if (!info.m_followerBody)
+        // The follower body has to be a RigidBody, otherwise it won't be moving anywhere
+        if (!info.m_followerBody || info.m_followerBody->GetNativeType() != NativeTypeIdentifiers::RigidBody)
```

Comment:
In the same way that inside the previous condition it does `info.m_leadBody = nullptr;` in the case it's not the right native type, should this do the same and assign `info.m_followerBody = nullptr` too in entered here?

Reply:
Good point, we're doing early return here, but still the `info` object lives outside the function, so need to null the invalid data. thanks!